lib/internal/modules/cjs/loader.js

```diff
@@ -297,26 +276,58 @@ function setModuleParent(value) {
   moduleParentCache.set(this, value);
 }
 
-ObjectDefineProperty(Module.prototype, 'parent', {
-  __proto__: null,
-  get: pendingDeprecation ? deprecate(
-    getModuleParent,
-    'module.parent is deprecated due to accuracy issues. Please use ' +
-      'require.main to find program entry point instead.',
-    'DEP0144'
-  ) : getModuleParent,
-  set: pendingDeprecation ? deprecate(
-    setModuleParent,
-    'module.parent is deprecated due to accuracy issues. Please use ' +
-      'require.main to find program entry point instead.',
-    'DEP0144'
-  ) : setModuleParent,
-});
-
 let debug = require('internal/util/debuglog').debuglog('module', (fn) => {
   debug = fn;
 });
-Module._debug = deprecate(debug, 'Module._debug is deprecated.', 'DEP0077');
+
+const builtinModules = [];
+function initializeCJS() {
+  const pendingDeprecation = getOptionValue('--pending-deprecation');
+  ObjectDefineProperty(Module.prototype, 'parent', {
+    __proto__: null,
+    get: pendingDeprecation ? deprecate(
+      getModuleParent,
+      'module.parent is deprecated due to accuracy issues. Please use ' +
+        'require.main to find program entry point instead.',
+      'DEP0144'
+    ) : getModuleParent,
+    set: pendingDeprecation ? deprecate(
+      setModuleParent,
+      'module.parent is deprecated due to accuracy issues. Please use ' +
+        'require.main to find program entry point instead.',
+      'DEP0144'
+    ) : setModuleParent,
+  });
+  Module._debug = deprecate(debug, 'Module._debug is deprecated.', 'DEP0077');
+
+  for (const { 0: id, 1: mod } of BuiltinModule.map) {
```

Comment:
is BuiltinModule.map safely iterable?

Reply:
This is still run before any user-land code, it was safely iterable before, I don't see why it's not now? If it needs fixing somehow, we probably shouldn't do this in a refactoring PR that's just moving code around.